### DISCUSSION TITLE
Is there any downside to using ts-node in dev env but tsc in production?

### DISCUSSION THREAD

**Discussion by rohit-malhotra** [TypeStrong/ts-node#1300] [Q&A] [answered]
Since the way typescript compilation works in tsc is different from ts-node, what are the downsides or possible gotchas that may happen due to this?

**Comment by cspotcode** [comment, marked as answer]
No big gotchas.  Generally the differences are in typechecking, which you can/should avoid by using `tsc` to typecheck and keeping `ts-node` fast with `--transpile-only`.  This is a personal recommendation; other people will disagree with me.  Ultimately, the decision is yours to make.  You can also avoid the major typechecking differences with `--files`.

`ts-node` forces a few compiler options to be enabled, overriding your tsconfig.json
https://github.com/TypeStrong/ts-node/blob/03355c15210731d948713a5edc624d4f215d1113/src/configuration.ts#L7-L17

These are pretty harmless changes.  For example, enabling sourcemaps does not affect the emitted JS code other than appending a `//#` sourcemap comment, and you may already have sourcemaps enabled.

We intentionally avoid deviating from `tsc` behavior much -- despite the community asking us to do so often -- because we want to remain pretty close in behavior to `tsc`.  It's not perfect, there are differences, but the goal is to be very close to `tsc` so that you can use `ts-node` as a development convenience and use `tsc` for production to reap the performance benefits of pre-compilation.

**Reply by cspotcode** [reply]
I forgot to mention, we also make these changes to your tsconfig.
https://github.com/TypeStrong/ts-node/blob/03355c15210731d948713a5edc624d4f215d1113/src/configuration.ts#L19-L42

As you can see, the deleted options should not affect the emitted `.js`, and you can explicitly specify `target` and `module` in your tsconfig to avoid our defaults.

**Reply by rafpaf** [reply]
Are these combinations equally safe?
* Test with `ts-node`, build with `tsc` 
* Test with `esbuild`, build with `tsc`
* Test with `swc`, build with `tsc`
